addons/hr_attendance/models/hr_attendance.py

```diff
@@ -92,71 +92,46 @@ def _compute_color(self):
-                                  as date)) = date_trunc('day', ot.date)
-                   AND att.employee_id = ot.employee_id
-                   AND att.employee_id IN %s
-                   AND ot.adjustment IS false
-              ORDER BY att.check_in DESC
-            ''', (tuple(self.employee_id.ids),))
-            a = self.env.cr.dictfetchall()
-            grouped_dict = dict()
-            for row in a:
-                if row['ot_id'] and row['att_wh']:
-                    if row['ot_id'] not in grouped_dict:
-                        grouped_dict[row['ot_id']] = {'attendances': [(row['att_id'], row['att_wh'])], 'overtime_duration': row['ot_d']}
-                    else:
-                        grouped_dict[row['ot_id']]['attendances'].append((row['att_id'], row['att_wh']))
-
-            for overtime in grouped_dict:
-                overtime_reservoir = grouped_dict[overtime]['overtime_duration']
-                if overtime_reservoir > 0:
-                    for attendance in grouped_dict[overtime]['attendances']:
-                        if overtime_reservoir > 0:
-                            sub_time = attendance[1] - overtime_reservoir
-                            if sub_time < 0:
-                                att_progress_values[attendance[0]] = 0
-                                overtime_reservoir -= attendance[1]
-                            else:
-                                att_progress_values[attendance[0]] = float(((attendance[1] - overtime_reservoir) / attendance[1]) * 100)
-                                overtime_reservoir = 0
-                        else:
-                            att_progress_values[attendance[0]] = 100
-                elif overtime_reservoir < 0 and grouped_dict[overtime]['attendances']:
-                    att_id = grouped_dict[overtime]['attendances'][0][0]
-                    att_progress_values[att_id] = overtime_reservoir
-                    negative_overtime_attendances[att_id] = True
-        for attendance in self:
-            if negative_overtime_attendances[attendance.id]:
-                attendance.overtime_hours = att_progress_values.get(attendance.id, 0)
+        day_starts = {att: self._get_day_start_and_day(att.employee_id, att.check_in) for att in self}
+        date_min, date_max = min(date for dummy, date in day_starts.values()), max(date for dummy, date in day_starts.values())
+        datetime_min, datetime_max = min(time for time, dummy in day_starts.values()), max(time for time, dummy in day_starts.values())
+
+        # Find at least all attendances that happen on the same day and for the same employee
+        # as an attendance in self
+        all_attendances = self | self.env['hr.attendance'].search(
+            [
+                ('employee_id', 'in', self.employee_id.ids),
+                ('check_in', '>=', datetime_min),
+                ('check_in', '<=', datetime_max + timedelta(days=1)),
+            ],
+        )
```

Comment:
You could already sort the recordset here to not have to do it later on
```suggestion
        all_attendances = self | self.env['hr.attendance'].search([
            ('employee_id', 'in', self.employee_id.ids),
            ('check_in', '>=', datetime_min),
            ('check_in', '<=', datetime_max + timedelta(days=1)),
        ], order='check_in desc')
```